src/tests/multihost/ipa/test_subid_ranges.py

```diff
@@ -0,0 +1,116 @@
+""" Automation of IPA bugs """
+
+import pytest
+import subprocess
+import time
+import os
+from sssd.testlib.common.utils import SSHClient
+
+
+test_password = "Secret123"
+user = 'admin'
+
+
+def execute_cmd(multihost, command):
+    """ Execute command on client """
+    cmd = multihost.client[0].run_command(command)
+    return cmd
+
+
+@pytest.mark.usefixtures('prepare_list_subid_ranges',
+                         'setup_ipa_client')
+@pytest.mark.tier1
+class TestSubid(object):
+    """
+    This is for ipa bugs automation
+    """
+    def test_subid_feature(self, multihost,
+                           create_bkp_for_subid_files):
+        """
+        :Title: support subid ranges managed by FreeIPA
+        :id: 50bcdc28-00c8-11ec-bef4-845cf3eff344
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1803943
+        :steps:
+            1. Generate subid for user admin
+            2. Test newuidmap command
+            3. Test newgidmap command
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+            3. Should succeed
+        """
+        ssh1 = SSHClient(multihost.client[0].sys_hostname,
+                         username=user, password=test_password)
+        (result, result1, exit_status) = ssh1.exec_command(f"ipa  "
+                                                           f"subid-find"
+                                                           f"  --owner  "
+                                                           f"{user}")
+        user_details = result1.readlines()
+        uid_start = int(user_details[5].split(': ')[1].split('\n')[0])
+        uid_range = int(user_details[6].split(': ')[1].split('\n')[0])
+        gid_start = int(user_details[7].split(': ')[1].split('\n')[0])
+        gid_range = int(user_details[8].split(': ')[1].split('\n')[0])
+        grep_bash = "ps -ef | grep bash"
+        cmd = execute_cmd(multihost, f"{grep_bash}").stdout_text
+        find_admin = f"{grep_bash} | grep {user}"
+        if user in cmd:
+            proces_id = [int(i) for i in
+                         execute_cmd(multihost,
+                                     find_admin).stdout_text.split()
+                         if i.isdigit()][0]
+            multihost.client[0].run_command(f'kill -9 {proces_id}')
+        (results1, results2, results3) = ssh1.exec_command('unshare -U bash')
+        proces_id = [int(i)
+                     for i in execute_cmd(multihost,
+                                          find_admin).stdout_text.split()
+                     if i.isdigit()][0]
+        ssh1.exec_command(f"newuidmap {proces_id}  {uid_start} "
+                          f"{uid_start + 1} 1")
+        ssh1.exec_command(f"newgidmap  {proces_id} {gid_start}"
+                          f" {gid_start + 1} 1")
+        result = execute_cmd(multihost, f"cat /proc/{proces_id}/uid_map")
+        assert f'{uid_start} {uid_start + 1}' in result.stdout_text
```

Comment:
Does `in` ignores white spaces / tabs?

Reply:
No it should be exactly the same line , as you have asked me to put entire result output . so i have changed it

Reply:
Example output is:
```
$ cat /proc/3938961/uid_map 
         0     100000          1
```

I don't think `f'{uid_start} {uid_start + 1}'` matches this without ignoring white spaces / tabs.

My request was to match all components of `uid loweruid count exactly, instead of making sure only one param is there (on unspecified position).

Reply:
Btw, while:
```
newuidmap {proces_id}  {uid_start} {uid_start + 1} 1
```
is valid usage it is quite misleading.

Man page:
```
       newuidmap pid uid loweruid count [uid loweruid count [ ... ]]
```
I would propose to use '0' as `uid` (quite typical application) and `count` could be any number between '1' and range size 'uid_range'

Reply:
fixed